packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -94,10 +94,28 @@ class ScriptLocation {
   String toString() => '${scriptRef.uri} $location';
 }
 
-/// A line, column, and an optional tokenPos.
 class SourcePosition {
-  SourcePosition({@required this.line, @required this.column, this.tokenPos});
+  const SourcePosition({
+    @required this.line,
+    @required this.column,
+    this.file,
+    this.tokenPos,
+  });
+
+  factory SourcePosition.calculatePosition(Script script, int tokenPos) {
+    final List<List<int>> table = script.tokenPosTable;
+    if (table == null) {
```

Comment:
since we aren't using the `table` var can we just check `if (script.tokenPosTable == null)` directly?